common/txmgr/confirmer.go

```diff
@@ -1004,28 +1024,37 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) han
 	}
 }
 
-// EnsureConfirmedTransactionsInLongestChain finds all confirmed txes up to the depth
+// EnsureConfirmedTransactionsInLongestChain finds all confirmed txes up to the LatestFinalizedHead
```

Comment:
I think the previous comment was probably also wrong but we're really finding all confirmed txes up to the earliest head in the given chain.